libraries/aws/provisioning/include/types/aws_iot_provisioning_types.h

```diff
@@ -277,6 +277,41 @@ typedef struct AwsIotProvisioningRejectedResponse
     size_t errorMessageLength;  /**< The length of the error messsage sent by the server. */
 } AwsIotProvisioningRejectedResponse_t;
 
+/**
+ * @ingroup provisioning_datatypes_paramstructs
+ * @brief Aggregates the data sent as response from AWS IoT Core for the request to create a new certificate for a CSR request.
+ *
+ * @paramfor Response Callback function of @ref provisioning_function_createkeysandcertificate
+ *
+ * The @ref AwsIotProvisioning_CreateKeysAndCertificate library API passes this object to a user-provided callback
+ * function
+ * whenever the operation completes with a response from the server.
+ */
+typedef struct AwsIotProvisioningCreateCertificateFromCsrResponse
+{
+    /** @brief The highest level HTTP based status code sent by the server. */
+    AwsIotProvisioningServerStatusCode_t statusCode;
+
+    union
+    {
+        /** @brief Represents the successful/accepted response of device credentials received from the server. */
+        struct
+        {
+            const char * pDeviceCertificate;         /**< The new certificate for the device.*/
+            size_t deviceCertificateLength;          /**< The size of the device certificate.*/
+            const char * pCertificateId;             /**< The certificate ID associated with the new certificate,
+                                                      * @p pDeviceCertificate.*/
+            size_t certificateIdLength;              /**< The length of the certificate ID.*/
+            const char * pCertificateOwnershipToken; /**< The token that represents ownership of certificate and
+                                                      * associated private key that the device.*/
+            size_t ownershipTokenLength;             /**< The size of the ownership token.*/
+        } acceptedResponse;
+
+        /** @brief Represents the rejected response information received from the server. */
+        AwsIotProvisioningRejectedResponse_t rejectedResponse;
+    } u; /**< @brief Valid member depends on operation status. */
+} AwsIotProvisioningCreateCertificateFromCsrResponse_t;
```

Comment:
These type names seem a bit long. Is there a problem with a shorter name like `AwsIotProvisioningCsrCreateCertResponse_t`?

Reply:
No issue, thanks for the suggestion :)